src/Microsoft.Restier.Core/ApiConfiguration.cs

```diff
@@ -60,6 +66,23 @@ public IServiceProvider ServiceProvider
         internal IEdmModel Model { get; private set; }
 
         /// <summary>
+        /// Adds a configuration procedure for API type <typeparamref name="T"/>.
+        /// </summary>
+        /// <typeparam name="T">The API type.</typeparam>
+        /// <param name="configurationCallback">
+        /// An action that will be called during the configuration of <typeparamref name="T"/>.
+        /// </param>
+        [CLSCompliant(false)]
+        public static void Configure<T>(Action<IServiceCollection> configurationCallback)
```

Comment:
I see. So as long as we call `ApiConfiguration.Configure<SomeApi>(do sth)` in the startup code, the callback will automatically get called the first time an `ApiConfiguration` instance for `SomeApi` is created. That may be a good solution for #323. Let's keep it and see how it works:-) Thanks for that!